openapi-validation-core/src/main/java/com/getyourguide/openapi/validation/core/OpenApiRequestValidator.java

```diff
@@ -42,11 +42,19 @@ public boolean isReady() {
     }
 
     public void validateRequestObjectAsync(final RequestMetaData request, String requestBody) {
-        threadPool.execute(() -> validateRequestObject(request, requestBody));
+        executeAsync(() -> validateRequestObject(request, requestBody));
     }
 
     public void validateResponseObjectAsync(final RequestMetaData request, ResponseMetaData response, final String responseBody) {
-        threadPool.execute(() -> validateResponseObject(request, response, responseBody));
+        executeAsync(() -> validateResponseObject(request, response, responseBody));
+    }
+
+    private void executeAsync(Runnable command) {
+        try {
+            threadPoolExecutor.execute(command);
+        } catch (RejectedExecutionException ignored) {
+            // ignored
```

Comment:
Do you want to monitor this? It could point to a configuration issue (thread pool too small for the amount of requests handled). If you are not monitoring, you could simply set up the In `ThreadPoolExecutor.DiscardPolicy` handler instead, see https://stackoverflow.com/questions/8183205/what-could-be-the-cause-of-rejectedexecutionexception